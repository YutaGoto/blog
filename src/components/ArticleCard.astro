---
interface Props {
  post: CollectionEntry<"posts">
}

import type { CollectionEntry } from "astro:content"
import { cdate } from "cdate"

import Tag from "./Tag.astro"

const { post } = Astro.props
const title = post.data.title
const tags = post.data?.tags
---

<div
  class="mb-2 bg-gray-50 px-4 py-10 shadow-lg dark:bg-gray-800 sm:rounded-xl sm:p-5"
  transition:name=`articleCard-${post.slug}`
  transition:animate="fade"
>
  <a
    href={`/posts/${post.slug}`}
    class="has-text-black"
    astro-data-history="push"
  >
    <article class="media" itemtype="http://schema.org/Article">
      <div class="">
        <div class="mb-2 text-2xl">{title}</div>
        <div class="mb-2 text-sm">
          {cdate(post.data.date).format("YYYY年MM月DD日")}
        </div>
        <div class="mb-2">
          {
            tags?.map((tag: string) => (
              <Tag className="mr-1" tag={tag} withLink={false} />
            ))
          }
        </div>
      </div>
    </article>
  </a>
</div>
