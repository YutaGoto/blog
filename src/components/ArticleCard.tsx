import React from "react"
import { Tag } from "./Tag"
import { cdate } from "cdate"
import type { MarkdownInstance } from "astro"

interface ArticleCardProps {
  post: MarkdownInstance<Record<string, any>>
}

export const ArticleCard = ({ post }: ArticleCardProps) => {
  const title = post.frontmatter.title
  const tags = post.frontmatter.tags

  return (
    <div className="mb-2 bg-gray-50 px-4 py-10 shadow-lg dark:bg-gray-800 sm:rounded-xl sm:p-5">
      <a href={post.url} itemProp="url" className="has-text-black">
        <article
          className="media"
          itemScope
          itemType="http://schema.org/Article"
        >
          <div className="">
            <div className="mb-2 text-2xl">{title}</div>
            <div className="mb-2 text-sm">
              {cdate(post.frontmatter?.date).format("YYYY年MM月DD日")}
            </div>
            <div className="mb-2">
              {tags?.map((tag: string) => (
                <Tag key={tag} className="mr-1" tag={tag} withLink={false} />
              ))}
            </div>
            <div
              className="text-base"
              dangerouslySetInnerHTML={{
                __html: post.frontmatter.description || "",
              }}
              itemProp="description"
            />
          </div>
        </article>
      </a>
    </div>
  )
}
