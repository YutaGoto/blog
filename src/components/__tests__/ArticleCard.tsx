import React from "react"
import { create } from "react-test-renderer"
import { render } from "@testing-library/react"
import { ArticleCard } from "../ArticleCard"

describe("ArticleCard", () => {
  const post = {
    excerpt: "こんにちは、 .ごっちです。",
    fields: {
      slug: "/20211223-imas-mokumoku/",
    },
    frontmatter: {
      date: "December 23, 2021",
      title: "アイマスもくもく会2021年まとめ",
      description: "2021年もアイマスもくもく会でしたよ",
      tags: ["アイマス"],
    },
  }

  it("renders correctly", () => {
    const tree = create(<ArticleCard post={post} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("display the div element", () => {
    const { container } = render(<ArticleCard post={post} />)
    expect(container.querySelector(`div`)).toBeInTheDocument()
  })
})
