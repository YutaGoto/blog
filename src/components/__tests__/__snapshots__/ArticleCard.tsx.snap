// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard renders correctly 1`] = `
<div
  className="px-4 py-10 mb-2 bg-gray-50 dark:bg-gray-800 shadow-lg sm:rounded-xl sm:p-5"
>
  <a
    className="has-text-black"
    href="/20211223-imas-mokumoku/"
    itemProp="url"
  >
    <article
      className="media"
      itemScope={true}
      itemType="http://schema.org/Article"
    >
      <div
        className=""
      >
        <div
          className="text-2xl mb-2"
        >
          アイマスもくもく会2021年まとめ
        </div>
        <div
          className="text-sm mb-2"
        >
          December 23, 2021
        </div>
        <div
          className="mb-2"
        >
          <span
            className="mr-1 py-1 px-2 rounded text-sm bg-emerald-800"
          >
            <a
              href="/tags/アイマス"
            >
              アイマス
            </a>
          </span>
        </div>
        <div
          className="text-base"
          dangerouslySetInnerHTML={
            Object {
              "__html": "2021年もアイマスもくもく会でしたよ",
            }
          }
          itemProp="description"
        />
      </div>
    </article>
  </a>
</div>
`;
